website/docs: FreeIPA documentation updates (#15183)
goauthentik/authentik: website/docs/users-sources/sources/directory-sync/freeipa/index.md

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.

All the settings are described in [LDAP sources](../../protocols/ldap/index.mdx).
:::

In authentik, create a new LDAP Source: go to **Directory** > **Federation and Social Login**, click **Create**,
select **LDAP source**.

These settings are required for FreeIPA: 

- **Name**: `Company FreeIPA`
- **Slug**: `company-freeipa`
- **Update internal password on login**: enable if you want your users to be able to login if FreeIPA is not accessible
- **Delete not found object**: enable to delete users from Authentik when they are deleted in FreIPA

### Connection settings

- **Server URI**: `ldaps://ipa1.freeipa.company`

    :::tip
    You can specify multiple servers by separating URIs with a comma, like `ldap://ipa1.freeipa.company,ldap://ipa2.freeipa.company`.

    When using a DNS entry with multiple Records, authentik will select a random entry when first connecting.
    :::

- **Enable StartTLS**: enable for `ldap://` protocol, disable for `ldaps://`

> - **TLS Verification Certificate**: **FIXME**

Reviewer: This needs to be completed

- **Bind CN**: `uid=svc_authentik,cn=users,cn=accounts,dc=freeipa,dc=company`

> - **Bind Password**: The password you've given the user above

Reviewer: ```suggestion
- **Bind Password**: The password for the above user account.
```

- **Base DN**: `dc=freeipa,dc=company`

### LDAP Attribute mapping

- **User Property Mappings**: Control/Command-select all Mappings which start with "authentik default LDAP" and "authentik default OpenLDAP". Remove others that are selected by default.
- **Group property mappings**: Select "authentik default OpenLDAP Mapping: cn"

### Additional settings

- **Parent Group**: If selected, all synchronized groups will be given this group as a parent.
- **User Path**: `company/freeipa` for instance, this setting is not crucial.
- **Addition User/Group DN**: `cn=users,cn=accounts`
- **Addition Group DN**: `cn=groups,cn=accounts`
- **User object filter**: `(objectClass=person)`
- **Group object filter**: `(objectClass=groupofnames)`
- **Group membership field**: `memberOf`
- **User membership attribute**: `distinguishedName`
- **Lookup using user attribute**: *enabled*.
- **Object uniqueness field**: `ipaUniqueID`

:::caution

> In FreeIPA, groups can have other groups as members. Indirect member users are not listed in the parent group's `member` attribute.
> Because of this, the sync havs to be done by the user's attribute `memberOf`. These are present for every membership, even indirect one.

Reviewer: ```suggestion
FreeIPA groups can contain nested groups. The `memberOf` user attribute lists all group memberships, direct and indirect.

If you want to sync only direct group memberships, use the following settings:
```
```

> In the improbable case that you want to sync only direct memberships, you can use the following settings:

Reviewer: ```suggestion
```
